Remove `add_reasons_to_defaults()` from general initialization

2ndkauboy opened the pull request:
The `add_reasons_to_defaults()` adds the translatable strings to the `$default`. Those `reasons` are only needed inside the `send_mail_notification()` method, which also calls the `add_reasons_to_defaults()` function. Therefore, the function call can be removed from the general `init()` function.

Fixes #625

2ndkauboy committed:
Remove `add_reasons_to_defaults()` from general initialization

The `add_reasons_to_defaults()` adds the translatable strings to the `$default`.
Those `reasons` are only needed inside the `send_mail_notification()` method,
which also calls the `add_reasons_to_defaults()` function. Therefore, the function
call can be removed from the general `init()` function.

2ndkauboy committed:
Remove `load_plugin_lang()` function and utilize `_load_textdomain_just_in_time()`

This function is calling the `load_plugin_textdomain()`, which eventually become
deprecated soon. Since this plugin is also using the "Text Domain" plugin header,
it is not necessary to call this function at all.

By removing this function, the `_load_textdomain_just_in_time()` function will be
used, when the first string needs to be translated.

swissspidy commented:
I think this re-introduces #508.

1. Go to the Antispam Bee settings page
2. See warnings

Query Monitor screenshot:

<img width="964" alt="Screenshot 2024-10-18 at 14 43 53" src="https://github.com/user-attachments/assets/cb93e90a-7aa9-4b9f-892e-c8bd0d32230b">

----

Translation-wise it looks fine though.

2ndkauboy committed:
Add an action to load the reasons on `init`

2ndkauboy commented:
@swissspidy great catch!

I first wondered on how/where the `Antispam_Bee_GUI` was able to get `self::$defaults`, but then I saw it: `... extends Antispam_Bee` :see_no_evil: 

Oh boy! Time to get this old code gone and `v3` ready!

My new commit should fix it. It now loads the "reasons" in an `init` hook callback. Still too much, but it should resolve both issues.

2ndkauboy committed:
Raise "Requires at least" version to 4.6

Since we don't explicitly load the translation files anymore
this will make sure that the necessary function is there.

2ndkauboy committed:
Fix PHPCS issues

stklcode reviewed (approved):
LGTM
